fix(formatter): Print comment in ternary jsx

leaysgur opened the pull request:
Fixes #16194

Just prints `self.expression`, instead of `FormatNodeWithoutTrailingComments`.

Aligned with Prettier 3.7.1 outputs:

```tsx
{
  isVideo ? (
    <Video {...props} />
  ) : (
    <Image {...props} /> // eslint-disable-line
  );
}

<>
  {isVideo ? (
    <Video {...props} />
  ) : (
    <Image {...props} /> // eslint-disable-line
  )}
</>;
```

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEwCWBnAamgJnCAAgH5CAKAHSkMIB4d8jgA6VgBwCcI2MBfQgPQA+KgEpCSclRq0AkgFsAhgHM4hFuy49+wwQMJwMAGzSwAtLkyKARkbhmTUOGN5UqtEVHTY8BElOo6Bj8NZk5uPkFPcUlKQLklVXVWMK1I3QF9Q0cYCytbe0dnKFFXKFphEAAaEG4YNGgMZFBFDi4AdwAFVoQmlEUjdsUATyaa6w5FMABrOBgAZUV5OAAZUzhkADMBjDgaiGsAKzgwGAB1SbZkEE5DOA4ANw3xyZm5+bYp02VkGA4AVz2IF28jQvwBQLgAA82Pc0MtYAMAPKwyYwCAcToQDBoerQa4IXDVEDQ1HwhAwAYAFXuUFaaEMWx2QJxUGUdgAiv8IPAmUZdjVDhgofNvpzubykNt+UCAI4SuCdNLXRQYMxOOD4Ik1P6KNCOZQAYQg8iUKqMRmJrPZcAAgjA-mhrP94J17msnHyBSAABYweRGM4+3GGT5gODzXq4tAPXHDa5gDBjEAPQGyKD4WDzMAcNBsGC2jPzGDDOxeoHhXYXRRXFC3XaPZ4gUwNmBKlRmqXMmqfDgN67WGxwS093OwM54GA+5AADgADDUOHB5Wgl+3lJ3pd7KdYJ7gp8gAEw1f67Kk2PpbyHyayarUrRRs-4qOAAMQxSgd3xVLogIF4vBAA

Dunqing commented on crates/oxc_formatter/src/utils/conditional.rs:
```diff
@@ -344,9 +344,9 @@ impl<'a> FormatConditionalLike<'a, '_> {
-        false
+    fn has_multiline_comment(&self, f: &Formatter<'_, 'a>) -> bool {
+        let span = self.span();
+        f.comments().has_comment_in_range(span.start, span.end)
```
It doesn't check any comments to see whether they have multiple lines. 

I don't understand the logic in [Biome](https://github.com/biomejs/biome/blob/ceed34a164cc76406baf617000b8877160f0bcc3/crates/biome_js_formatter/src/utils/conditional.rs#L197-L219), and no tests fail by this, so I didn't implement it. Can you double what it does? Is there any case that would be affected by this?

Dunqing replied:
Take a look at Prettier as well, maybe it is already dead code

Dunqing replied:
Remove them then! File an issue about this in Prettier if you'd like.

leaysgur replied:
OK, I'll:

- Revert some useless change here to focus fixing bug
- Remove `has_multiline_comment` and related lines
- Report to prettier

leaysgur committed:
fix(formatter): Print comment in ternary jsx (#16224)

Fixes #16194

Just prints `self.expression`, instead of `FormatNodeWithoutTrailingComments`.

Aligned with Prettier 3.7.1 outputs:

```tsx
{
  isVideo ? (
    <Video {...props} />
  ) : (
    <Image {...props} /> // eslint-disable-line
  );
}

<>
  {isVideo ? (
    <Video {...props} />
  ) : (
    <Image {...props} /> // eslint-disable-line
  )}
</>;
```

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEwCWBnAamgJnCAAgH5CAKAHSkMIB4d8jgA6VgBwCcI2MBfQgPQA+KgEpCSclRq0AkgFsAhgHM4hFuy49+wwQMJwMAGzSwAtLkyKARkbhmTUOGN5UqtEVHTY8BElOo6Bj8NZk5uPkFPcUlKQLklVXVWMK1I3QF9Q0cYCytbe0dnKFFXKFphEAAaEG4YNGgMZFBFDi4AdwAFVoQmlEUjdsUATyaa6w5FMABrOBgAZUV5OAAZUzhkADMBjDgaiGsAKzgwGAB1SbZkEE5DOA4ANw3xyZm5+bYp02VkGA4AVz2IF28jQvwBQLgAA82Pc0MtYAMAPKwyYwCAcToQDBoerQa4IXDVEDQ1HwhAwAYAFXuUFaaEMWx2QJxUGUdgAiv8IPAmUZdjVDhgofNvpzubykNt+UCAI4SuCdNLXRQYMxOOD4Ik1P6KNCOZQAYQg8iUKqMRmJrPZcAAgjA-mhrP94J17msnHyBSAABYweRGM4+3GGT5gODzXq4tAPXHDa5gDBjEAPQGyKD4WDzMAcNBsGC2jPzGDDOxeoHhXYXRRXFC3XaPZ4gUwNmBKlRmqXMmqfDgN67WGxwS093OwM54GA+5AADgADDUOHB5Wgl+3lJ3pd7KdYJ7gp8gAEw1f67Kk2PpbyHyayarUrRRs-4qOAAMQxSgd3xVLogIF4vBAA
